content/documentation/pages/6-recipes/7-scaling/0-index.md

```diff
@@ -0,0 +1,29 @@
+
+In Spring Cloud Data Flow the [stream data processing](%currentPath%/concepts/streams/) is implemented as a collection of independent event-driven streaming applications (microservices) that connect using a messaging middleware such as RabbitMQ, Apache Kafka.
+The ability of a streaming pipeline to cope with increased data load depends on the:
+
+- _Messaging Middleware Scalability_ - [data partitioning](https://docs.spring.io/spring-cloud-stream/docs/current/reference/htmlsingle/#partitioning) is a widely used technique for scaling messaging middleware and Data Flow provides extensive [stream partition support](%currentPath%/feature-guides/streams/partitioning/#data-partitioning/).
+- _Event-driven Applications Scalability_ - Spring Cloud Data Flow provides event-driven applications parallelism, referred to as application scaling. Later allows to allocate multiple application instances, usually one for every data partition topic.
+
+Following diagram illustrates a typical scale-out architecture based on data partitioning and applications parallelism:
+
+![SCDF Stream Scaling](images/scdf-stream-partition-and-scaling-overview.png)
+
+Platforms such as Kubernetes and Cloud Foundry offer scaling features for the TechOps teams to control system’s load. For example TechOps can leverage [CF Scale](https://docs.cloudfoundry.org/devguide/deploy-apps/cf-scale.html), [scale applications](https://docs.spring.io/spring-cloud-dataflow/docs/2.3.0.RELEASE/reference/htmlsingle/#configuration-cloudfoundry-scaling) or [Kubectl Scale](https://jamesdefabia.github.io/docs/user-guide/kubectl/kubectl_scale/) to scale applications manually or enable autoscaling features such as [App Autoscaler](https://docs.run.pivotal.io/appsman-services/autoscaler/using-autoscaler.html) for Cloud Foundry and [HPA](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) and [KEDA](https://github.com/kedacore/keda) for Kubernetes. The autoscaling is usually controlled by CPU or memory limit thresholds (HPA) or message queues sizes (KEDA).
+
+While the scaling is happens outside of the Spring Cloud Data Flow life-cycle, later gracefully handles the application scale changes. One only need to configure the message partitioning using properties such as [partitionKeyExpression](https://cloud.spring.io/spring-cloud-static/spring-cloud-stream/3.0.0.RELEASE/reference/html/spring-cloud-stream.html#spring-cloud-stream-overview-configuring-output-bindings-partitioning) and [partitionCount](https://cloud.spring.io/spring-cloud-static/spring-cloud-stream/3.0.0.RELEASE/reference/html/spring-cloud-stream.html#spring-cloud-stream-overview-configuring-output-bindings-partitioning).
+
+In addition to the platform specific, low-level APIs, Spring Cloud Data Flow provides a dedicated, fine-grained Scale API, designed for scaling data pipelines. Later provides an uniform way to leverage the platform scaling capabilities from within custom Spring applications.
```

Comment:
Why does it need to be defined as fine-grained?

Reply:
Because in my opinion it is tuned particularly for the  SCDF stream apps (and actually any app that SCDF can deploy).  But maybe `dedicated` already says this?